api/librdb-api.h

```diff
@@ -155,26 +174,42 @@ typedef struct RdbHandlersRawCallbacks {
     HANDLERS_COMMON_CALLBACKS
     RdbRes (*handleStringValue)(RdbParser *p, void *userData, RdbBulk str);
-    RdbRes (*handlerQListNode)(RdbParser *p, void *userData, RdbBulk listNode);
-    RdbRes (*handlerPlainNode)(RdbParser *p, void *userData, RdbBulk node);
-    /* ... TBD ... */
+    RdbRes (*handleQListNode)(RdbParser *p, void *userData, RdbBulk listNode);
+    RdbRes (*handlePlainNode)(RdbParser *p, void *userData, RdbBulk node);
+
+    /*** TODO: RdbHandlersStructCallbacks: handlerHashListPack, handleSet, handleZsetListPack, handleFunction, handleStreamListPack, handleStreamMetadata, handleStreamCgroup ***/
+    RdbRes (*handlerHashListPack)(RdbParser *p, void *userData, RdbBulk hash);
+    RdbRes (*handleSet)(RdbParser *p, void *userData, RdbBulk intSet);
+    RdbRes (*handleZsetListPack)(RdbParser *p, void *userData, RdbBulk zset);
+    RdbRes (*handleFunction)(RdbParser *p, void *userData, RdbBulk func);
+    RdbRes (*handleStreamListPack)(RdbParser *p, void *userData, RdbBulk nodekey, RdbBulk listpack);
+    RdbRes (*handleStreamMetadata)(RdbParser *p, void *userData, uint64_t len, uint64_t entriesAdded, streamID *firstID, streamID *lastID, streamID *maxDeletedEntryID);
+    RdbRes (*handleStreamCgroup)(RdbParser *p, void *userData, streamCG *cgroup); /* when succeeds, 'cgroup' belongs to the calee  */
+
 } RdbHandlersStructCallbacks;
 
 typedef struct RdbHandlersDataCallbacks {
     HANDLERS_COMMON_CALLBACKS
     RdbRes (*handleStringValue)(RdbParser *p, void *userData, RdbBulk str);
     RdbRes (*handleListElement)(RdbParser *p, void *userData, RdbBulk str);
-//    RdbRes (*handleSetElement)(RdbParser *p, void *userData, RdbBulk str, unsigned long sizeHint);
-//    RdbRes (*handleZsetElement)(RdbParser *p, void *userData, RdbBulk str, double score, unsigned long sizeHint);
-//    RdbRes (*handleHashElement)(RdbParser *p, void *userData, RdbBulk field, RdbBulk value, unsigned long sizeHint);
-//    RdbRes (*handleModuleDatatype)(RdbParser *p, void *userData, RdbBulk value);
-    /* ... TBD ... */
+
+    /*** TODO: RdbHandlersDataCallbacks: handleHashElement, handleSetElement, handleZsetElement, handleStreamElement, handleStreamMetadata, handleStreamCgroup  ***/
+    RdbRes (*handleHashElement)(RdbParser *p, void *userData, RdbBulk field, RdbBulk elm, uint64_t totalNumElm);
+    RdbRes (*handleSetElement)(RdbParser *p, void *userData, RdbBulk elm, uint64_t totalNumElm);
+    RdbRes (*handleZsetElement)(RdbParser *p, void *userData, RdbBulk elm, double score, uint64_t totalNumElm);
+    RdbRes (*handleStreamElement)(RdbParser *p, void *userData, streamID *id, RdbBulk field, RdbBulk value, uint64_t totalNumElm);
+    RdbRes (*handleStreamMetadata)(RdbParser *p, void *userData, uint64_t len, uint64_t entriesAdded, streamID *firstID, streamID *lastID, streamID *maxDeletedEntryID);
+    RdbRes (*handleStreamCgroup)(RdbParser *p, void *userData, streamCG *cgroup); /* when succeeds, 'cgroup' belongs to the calee  */
```

Comment:
again, i think we must break these into separate smaller CBs. otherwise, as soon as some new field is added, we won't be API or ABI compatible.